chore: add GitHub metadata

EndBug opened the pull request:
This PR adds a citation file and a license to the repository.
While this info was already available in one form or another, adding it using the standard files not only makes it more accessible, but also allows GitHub to display it more promptly.

I created the citation file with the info I could find, so it's not really complete. You can also add your ORCIDs!

Closes #1006

pabloogc commented:
Thanks for creating the citation, I created my ORCID if you can update the PR: https://orcid.org/0009-0008-0080-1437

danielgallegovico commented:
I wasn't familiar to ORCID, but it's an interesting way of normalizing citations, thanks for proposing it! Can you include my ID in your PR? https://orcid.org/0009-0006-8582-4384

danielgallegovico commented on CITATION.cff:
```diff
@@ -0,0 +1,22 @@
+    email: ivanmartit@gmail.com
+  - family-names: Gallego Vico
+    given-names: Daniel
+    email: dgallego@dit.upm.es
```
```suggestion
    email: danielgallegovico@gmail.com
```

imartinez commented:
Thanks @EndBug . Here is my ORCID https://orcid.org/0009-0004-5065-2311

pabloogc commented on CITATION.cff:
```diff
@@ -0,0 +1,22 @@
+    given-names: Daniel
+    email: dgallego@dit.upm.es
+  - given-names: Pablo
+    family-names: Orgaz
```
```suggestion
    family-names: Orgaz
    email: pabloogc+gh@gmail.com
```

EndBug commented:
Hey folks, I've updated the citation files with the correct emails and your ORCIDs 🎉 

I've also updated the README so that it shows the BibTeX and APA that get generated from the citation file, so that it's not in conflict ;)
